src/Services/MarketTradingDataService.cs

```diff
@@ -19,75 +19,127 @@ public MarketTradingDataService(ICandlesHistoryServiceProvider candlesHistorySer
+                                 ?? 0;
 
-            return new AssetPairTradingData(assetPair, lastTradePrice, volume24H);
+            return new AssetPairTradingDataItem<double>(assetPair, lastTradePrice);
         }
 
-        public async Task<IEnumerable<AssetPairTradingData>> GetAllPairsAsync()
+        public async Task<IEnumerable<AssetPairTradingDataItem<double>>> TryGetAllPairsVolumeAsync()
         {
-            var spotCandlesTask = GetLastDayCandlesAsync(MarketType.Spot);
-            var mtCandlesTask = GetLastDayCandlesAsync(MarketType.Mt);
-
-            var spotCandles = await spotCandlesTask;
-            var mtCandles = await mtCandlesTask;
+            var candles = await GetLastDayCandlesAsync(); // Item1 = spot, Item2 = MT
 
-            var result = new Dictionary<string, AssetPairTradingData>();
+            var result = new Dictionary<string, AssetPairTradingDataItem<double>>();
 
-            foreach (var spotAssetCandles in spotCandles)
+            foreach (var spotAssetCandles in candles.Item1)
             {
                 var volume24 = spotAssetCandles.Value.History.Sum(c => c.TradingOppositeVolume);
 
-                if (mtCandles.TryGetValue(spotAssetCandles.Key, out var mtAssetCandles))
+                if (candles.Item2.TryGetValue(spotAssetCandles.Key, out var mtAssetCandles))
                 {
                     volume24 += mtAssetCandles.History.Sum(c => c.TradingOppositeVolume);
                 }
 
-                var assetPairData = new AssetPairTradingData(
+                var assetPairDataItem = new AssetPairTradingDataItem<double>(
                     spotAssetCandles.Key,
-                    spotAssetCandles.Value.History.LastOrDefault()?.LastTradePrice ?? 0,
                     volume24);
 
-                result.Add(assetPairData.AssetPair, assetPairData);
+                result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
             }
 
-            foreach (var mtAssetCandles in mtCandles)
+            foreach (var mtAssetCandles in candles.Item2)
             {
                 if (!result.ContainsKey(mtAssetCandles.Key))
                 {
-                    var assetPairData = new AssetPairTradingData(
+                    var assetPairDataItem = new AssetPairTradingDataItem<double>(
                         mtAssetCandles.Key,
-                        mtAssetCandles.Value.History.LastOrDefault()?.LastTradePrice ?? 0,
                         mtAssetCandles.Value.History.Sum(c => c.TradingOppositeVolume));
 
-                    result.Add(assetPairData.AssetPair, assetPairData);
+                    result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
                 }
             }
 
             return result.Values;
         }
 
+        public async Task<IEnumerable<AssetPairTradingDataItem<double>>> TryGetAllPairsLastTradePriceAsync()
+        {
+            var candles = await GetLastDayCandlesAsync(); // Item1 = spot, Item2 = MT
```

Comment:
To calculate last trade price we don't need to request entire day by hours, any single candle will be enough